src/inference/graph_inference.cpp

```diff
@@ -1740,7 +1742,7 @@ void GraphInference::PrepareForInference() {
     double feature_weight = factor_features_[hash];
     Factor visited_labels;
     std::shared_ptr<std::pair<double, Factor>> factor_feature_shared_pointer = std::make_shared<std::pair<double, Factor>>(feature_weight, f);
-    best_factor_features_first_level_[f.size()].InsertFactorFeature(factor_feature_shared_pointer, f, 0, FLAGS_maximum_depth, -1, visited_labels, FLAGS_factors_limit);
```

Comment:
n.b. the factors_limit flag is the beam size used for the factor features during inference